File: dataset/output-bk/MISTRAL_codestral-latest/Task169_MISTRAL_codestral-latest.java
import java.util.ArrayList;
import java.util.List;
import java.util.TreeSet;

public class Solution {
    public List<Integer> countSmaller(int[] nums) {
        List<Integer> result = new ArrayList<>();
        TreeSet<Integer> sortedSet = new TreeSet<>();
        for (int i = nums.length - 1; i >= 0; i--) {
            int index = sortedSet.headSet(nums[i]).size();
            result.add(index);
            sortedSet.add(nums[i]);
        }
        Collections.reverse(result);
        return result;
    }
}

File: dataset/output-bk/MISTRAL_codestral-latest/Task169_MISTRAL_codestral-latest.c
#include <stdlib.h>
#include <stdio.h>
#include <set>
#include <vector>
#include <algorithm>

int* countSmaller(int* nums, int numsSize, int* returnSize) {
    int* result = (int*)malloc(numsSize * sizeof(int));
    std::multiset<int> sortedSet;
    for (int i = numsSize - 1; i >= 0; i--) {
        auto it = sortedSet.upper_bound(nums[i]);
        int index = std::distance(sortedSet.begin(), it);
        result[i] = index;
        sortedSet.insert(nums[i]);
    }
    *returnSize = numsSize;
    return result;
}